include/itkBackscatterImageFilter.hxx

```diff
@@ -0,0 +1,158 @@
+void
+BackscatterImageFilter<TInputImage, TOutputImage>::BeforeThreadedGenerateData()
+{
+  Superclass::BeforeThreadedGenerateData();
+
+  // Initialize metric image
+  this->GetOutput()->Allocate();
+
+  // Initialize iVars used in ComputeBackscatter()
+  float nyquistFrequency = m_SamplingFrequencyMHz / 2;
+  float numComponents = this->GetInput()->GetNumberOfComponentsPerPixel();
+  m_FrequencyDelta = nyquistFrequency / numComponents;
+  m_StartComponent = m_FrequencyBandStartMHz / m_FrequencyDelta;
+  m_EndComponent = m_FrequencyBandEndMHz / m_FrequencyDelta;
+  if (m_EndComponent == 0) // If m_FrequencyBandEndMHz is not set
+  {
+    m_EndComponent = numComponents - 1; // Use all components
+  }
+  m_ConsideredComponents = m_EndComponent - m_StartComponent + 1;
+}
+
+template <typename TInputImage, typename TOutputImage>
+void
+BackscatterImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateData(const OutputRegionType & regionForThread)
+{
+  if (regionForThread.GetNumberOfPixels() == 0)
+  {
+    return;
+  }
+
+  const InputImageType * input = this->GetInput();
+  OutputImageType *      output = this->GetOutput();
+
+  ImageScanlineConstIterator<TInputImage> it(input, regionForThread);
+  it.GoToBegin();
+
+  // do the work
+  while (!it.IsAtEnd())
+  {
+    while (!it.IsAtEndOfLine())
+    {
+      InputIndexType index = it.GetIndex();
+
+      OutputPixelType estimatedBackscatter = ComputeBackscatter(index);
+
+      output->SetPixel(index, estimatedBackscatter);
+      ++it;
+    }
+    it.NextLine();
+  }
+};
+
+template <typename TInputImage, typename TOutputImage>
+typename BackscatterImageFilter<TInputImage, TOutputImage>::OutputPixelType
+BackscatterImageFilter<TInputImage, TOutputImage>::ComputeBackscatter(const InputIndexType & index) const
+{
+  using ScalarType = typename NumericTraits<OutputPixelType>::ValueType;
+
+  // Get RF spectra frequency bins at start and end pixel positions
+  auto           input = this->GetInput();
+  InputPixelType sample = input->GetPixel(index);
+  ScalarType     sum = 0;
+
+  Eigen::Matrix<float, Eigen::Dynamic, 2> A(m_ConsideredComponents, 2);
+  Eigen::Matrix<float, Eigen::Dynamic, 1> b(m_ConsideredComponents);
+  for (unsigned i = 0; i < m_ConsideredComponents; i++)
+  {
+    A(i, 0) = 1;
+    A(i, 1) = (1 + i + m_StartComponent) * m_FrequencyDelta; // x_i = frequency
+    b(i) = sample[i + m_StartComponent];                     // y_i = intensity
+    sum += sample[i + m_StartComponent];
+  }
+
+  if (m_EstimateType == 0)
+  {
+    return sum / m_ConsideredComponents;
+  }
+
+  // from https://eigen.tuxfamily.org/dox/group__LeastSquares.html
+  Eigen::Matrix<float, 1, 2> lineFit = A.householderQr().solve(b);
+  ScalarType                 frequencySlope = -lineFit(1);
+  ScalarType                 frequencyIntercept = lineFit(0);
+
+  switch (m_EstimateType)
+  {
+    case 0: // handled above for efficiency (avoids unnecessary line fitting)
```

Comment:
Since `case 0` is already handled above and not expected here, `case 0` should break into `default` rather than into `case 1`